src/js/compat.ts

```diff
@@ -274,3 +273,35 @@ export async function loadLockFile(lockFileURL: string): Promise<any> {
     return await response.json();
   }
 }
+
+/**
+ * Calculate the directory name of the current module.
+ * This is used to guess the indexURL when it is not provided.
+ */
+export async function calculateDirname(): Promise<string> {
+  if (IN_NODE_COMMONJS) {
+    return __dirname;
+  }
+
+  if (IN_NODE_ESM) {
+    const nodePath = await import("path");
+    const nodeUrl = await import("url");
```

Comment:
Do we need webpack ignores here or other nonsense like that?

Reply:
Good point. I don't think it is required to use `/* webpackIgnore */` for builtin modules.